pkg/controller/controller.go

```diff
@@ -334,6 +343,36 @@ func ttls() []string {
 	return []string{"15m", "30m", "1h", "2h", "4h"}
 }
 
-func categories() []string {
-	return []string{"explanation"}
+func (c *Controller) categories() []string {
+	validators := c.p.Validators()
+	categories := make([]string, 0, len(validators))
+	for v := range validators {
+		categories = append(categories, v)
+	}
+	categories = append(categories, "explanation")
+	return categories
+}
+
+// categoryPairs returns a list of Pairs for rendering.
+func (c *Controller) categoryPairs() []Pair {
+	pairs := make([]Pair, 0, len(c.categories()))
+	for _, category := range c.categories() {
+		pairs = append(pairs, Pair{
```

Comment:
```suggestion
func (c *Controller) categoryPairs() []*Pair {
	pairs := make([]Pair, 0, len(c.categories()))
	for _, category := range c.categories() {
		pairs = append(pairs, &Pair{
```

Reply:
Cannot make this change unfortunately since []Pair is required in the place below to pass to the html

https://github.com/abcxyz/jvs/blob/c74e145411363aed1d245d5530ed92f246df1abe/pkg/controller/controller.go#L297

Reply:
We can change that here: https://github.com/abcxyz/jvs/blob/c74e145411363aed1d245d5530ed92f246df1abe/pkg/controller/controller.go#L53

Reply:
Resolving this comment thread now since we no longer use Pair after #303